flocker/node/agents/cinder.py

```diff
@@ -38,23 +45,73 @@
 # a volume.
 DATASET_ID_LABEL = u'flocker-dataset-id'
 
-LOCAL_IPS = Field(
-    u"local_ips",
-    repr,
-    u"The IP addresses found on the target node."
-)
 
-API_IPS = Field(
-    u"api_ips",
-    repr,
-    u"The IP addresses and instance_ids for all nodes."
-)
+def _openstack_logged_method(method_name, original_name):
+    """
+    Run a method and log additional information about any exceptions that are
+    raised.
 
-COMPUTE_INSTANCE_ID_NOT_FOUND = MessageType(
-    u"blockdevice:cinder:compute_instance_id:not_found",
-    [LOCAL_IPS, API_IPS],
-    u"Unable to determine the instance ID of this node.",
-)
+    :param str method_name: The name of the method of the wrapped object to
+        call.
+    :param str original_name: The name of the attribute of self where the
+        wrapped object can be found.
+
+    :return: A function which will call the method of the wrapped object and do
+        the extra exception logging.
+    """
+    def _run_with_logging(self, *args, **kwargs):
+        original = getattr(self, original_name)
+        method = getattr(original, method_name)
+        with OPENSTACK_ACTION(operation=[method_name, args, kwargs]):
```

Comment:
These arguments are ``IBlockDeviceAPI`` arguments, which are in our control.
You are right that some of them might not be JSON-seriazable (like ``BlockDeviceVolume``, as pointed out by @wallrj in EBS logging branch review).
Filed  https://clusterhq.atlassian.net/browse/FLOC-2054 to track this issue.